### PR TITLE
[interp] remove InterpMethod entry in hash table on free_method callback

### PR DESCRIPTION
**Pull request by lewurm**
(from #5425, in order to get work upstream)

### CODE REVIEW

File: mono/mini/mini-runtime.c [hunk 1]
```diff
@@ -2212,9 +2212,18 @@ mono_jit_free_method (MonoDomain *domain, MonoMethod *method)
 	gboolean destroy = TRUE;
 	GHashTableIter iter;
 	MonoJumpList *jlist;
+	MonoJitDomainInfo *info = domain_jit_info (domain);
 
 	g_assert (method->dynamic);
 
+#ifdef ENABLE_INTERPRETER
+	if (mono_use_interpreter) {
```
Comment (vargaz) [inline mono/mini/mini-runtime.c]: This doesn't need an #ifdef.

File: mono/mini/mini-runtime.c [hunk 2]
```diff
@@ -2212,9 +2212,18 @@ mono_jit_free_method (MonoDomain *domain, MonoMethod *method)
...
+		mono_domain_jit_code_hash_lock (domain);
+		mono_internal_hash_table_remove (&info->interp_code_hash, method);
+		mono_domain_jit_code_hash_unlock (domain);
+	}
```
Comment (vargaz) [inline mono/mini/mini-runtime.c]: Shouldn't this free the InterpMethod structure as well ? Or mention that its domain mempool allocated.

Comment (lewurm) [reply mono/mini/mini-runtime.c]: it's mempool allocated; comment added.

**Comment by lewurm** [comment]
@monojenkins build Linux AArch64

**Comment by lewurm** [comment]
@monojenkins build Linux AArch64 Interpreter

**Comment by lewurm** [comment]
@monojenkins Linux ARMv7 hard float Interpreter

**Comment by lewurm** [comment]
@monojenkins build Linux ARMv5 soft float

**Comment by lewurm** [comment]
@monojenkins build Linux ARMv7 hard float

**Comment by lewurm** [comment]
@monojenkins build OS X x64

**Comment by lewurm** [comment]
@monojenkins build Linux ARMv7 hard float Interpreter

**Pull request merged**